samples/placeholder-accessors/placeholder-accessors.cpp

```diff
@@ -0,0 +1,99 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2018 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  placeholder-accessors.cpp
+ *
+ *  Description:
+ *    Sample code that illustrates how to use placeholder accessors.
+ *
+ **************************************************************************/
+
+#include <CL/sycl.hpp>
+
+#include <iostream>
+
+using namespace cl::sycl;
+
+namespace {
+class Worker;
+}
+
+class Doubler {
+  /* These are placeholder accessors so we can default construct Doubler, these
+   * accessors do not have a buffer associated with them yet. It will be
+   * attached during the command group */
+  accessor<const uint8_t, 1, access::mode::read, access::target::global_buffer,
+           access::placeholder::true_t>
+      in_accessor;
+  accessor<uint8_t, 1, access::mode::discard_write,
+           access::target::global_buffer, access::placeholder::true_t>
+      out_accessor;
+
+ public:
+  void perform_doubling(const uint8_t* input, uint8_t* output,
+                        size_t items) const {
```

Comment:
> I don't really think this method is const, since it binds an accessor, writes to it, etc. etc. I guess from the compiler's perspective it might be, but it's a little misleading.

Apparently the `const`-qualified overloads for `accessor::operator[]` permit writes (this is Not Good™️). I agree that `perform_doubling` shouldn't be `const`-qualified.